dev/PushNotifications/GetRawNotificationEventArgs.h

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#pragma once
+#include "PushNotificationReceivedEventArgs.h"
+#include "externs.h"
+
+static PCWSTR c_pushContractId = L"Windows.Push";
+
+namespace winrt::Microsoft::Windows::PushNotifications
+{
+    static winrt::Windows::Foundation::IInspectable Deserialize(winrt::Windows::Foundation::Uri const&)
+    {
+        {
+            auto lock = g_lock.lock();
+            if (g_activatedEventArgs)
+            {
+                return g_activatedEventArgs;
+            }
+
+            THROW_HR_IF_NULL_MSG(E_UNEXPECTED, g_waitHandleForArgs, "PushNotificationManager::RegisterActivator has not been called.");      
+        }
+
+        if (WaitForSingleObject(g_waitHandleForArgs.get(), 1000) == WAIT_OBJECT_0)
```

Comment:
>if (WaitForSingleObject(g_waitHandleForArgs.get(), 1000) == WAIT_OBJECT_0) [](http://example.com/codeflow?start=8&length=74)

So you only get one second to deserialize? This timeout seems surprising here.

Should returning the global arguments clear the global instance?

Reply:
I've been exercising this function, and it just takes a few microseconds to retrieve the event args. Ideally we should have some telemetry to determine a more accurate average processing time. Will increase to 2 seconds to fall on the safe side.

We should not clear the global event args, I don't think GetActivatedEventArgs() is a free-after-use function.